vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,27 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+// HealthcheckResult stores information about a single run of a healthcheck probe
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting, -1=error running probe
+	Output     string // Output from last check
+}
+
+// Health states
+const (
+	Starting  = "starting"  // Container is not yet ready
```

Comment:
Assuming `Retries=1`, the state after one failure would be `unhealthy`.

Usually, a healthcheck does not need to know its previous state in order to report its current state.

A possible exception might be a very simple probe that detects failure, but wants to report "starting" if it's currently still in the starting state (because it can't distinguish between starting and unhealthy). However, it can get this behaviour by returning exit code 2 in all cases, since "starting" is treated as unhealthy if the container wasn't already starting.